### ISSUE TITLE
No "ImportExportTools NG" menu added (v12.0.2)

### ISSUE THREAD

**Issue by tnilzon** [thundernest/import-export-tools-ng#357]
No "ImportExportTools NG" menu added in Thunderbird. It seems to install without any errors, but it is not possible to reach any import/export options.

ImportExportTools NG v12.0.2
Thunderbird v102.0.3 (64-bit)

System:
Operating System: Manjaro Linux
KDE Plasma Version: 5.26.2
KDE Frameworks Version: 5.99.0
Qt Version: 5.15.6
Kernel Version: 5.18.19-3-MANJARO (64-bit)
Graphics Platform: X11
Processors: 8 × Intel® Core™ i7-9700K CPU @ 3.60GHz
Memory: 31,2 GiB of RAM
Graphics Processor: Mesa Intel® UHD Graphics 630
Manufacturer: Gigabyte Technology Co., Ltd.
Product Name: Z390 I AORUS PRO WIFI

**Comment by tnilzon** [comment]
[UPDATE]: Suddenly (after adding some more add-ons and TB restarting) the "ImportExportTools NG" options started to appear in TB menus. Don´t know what happened, but for now it seems to work!

**Comment by tnilzon** [comment]
[UPDATE 2]: The issue is back. No more "ImportExportTools NG" options in the TB menus. Seems to be some random behavior, but now I can't get it to work.

**Comment by cleidigh** [comment]
@tnilzon 
Have you tried reinstalling?
Also have you looked at the console log.
@cleidigh

**Comment by tnilzon** [comment]
@cleidigh thanks for comment.
I tried reinstalling, but I can't say I noticed any improvement. But as for now it appears working again. I was also able to try "Export profile" and it worked perfectly.

**Comment by ercek** [comment]
I am having the same issue. I downloaded my Takeout mbox file and am trying to use Thunderbird for the first time. I have the latest version (downloaded today) and added this add-on through Thunderbird's built-in add-on but the importexport option does not appear on the Tools list. I restarted, uninstalled, and reinstalled manually using the gear. Still no luck. I didn't add any other add-ons as none looked useful to me, but don't think that should be required. Help appreciated! (I also see that the latest version of Thunderbird looks very different from past, including where the menus are; not sure if something 'broke' when creating this latest version.)

**Comment by cleidigh** [comment]
@ercek 
Are you on Linux also?
So IETNG installs a menu under the Tools main top menubar, but not the hamburger menu.
Also the context menus for a message or folder.
Are you seeing any?
If not can you check the console output.
@cleidigh

**Comment by ercek** [comment]
No, I am on a microsoft surface running the latest windows. I can provide the info if you tell me what is most important to know for this.
From the console, I have no idea how to interpret it so am pasting here. 
1668289480686	addons.xpi	WARN	Checking C:\Program Files\Mozilla Thunderbird\distribution\extensions for addons
This page is in Quirks Mode. Page layout may be impacted. For Standards Mode use “<!DOCTYPE html>”.
[hiddenWindow.html](resource://gre-resources/hiddenWindow.html)
Successfully loaded OpenPGP library rnp.dll version 0.16+git20220124.f06439f7.MZLA from C:\Program Files\Mozilla Thunderbird\rnp.dll [RNPLib.jsm:100:15](chrome://openpgp/content/modules/RNPLib.jsm)
Found 0 public keys and 0 secret keys (0 protected, 0 unprotected) [RNPLib.jsm:301:15](chrome://openpgp/content/modules/RNPLib.jsm)
services.settings: Failed to load last_modified.json: TypeError: NetworkError when attempting to fetch resource. [Utils.jsm:330](resource://services-settings/Utils.jsm)
This page is in Quirks Mode. Page layout may be impacted. For Standards Mode use “<!DOCTYPE html>”.
[background.html](moz-extension://6a823ae8-977b-45cc-b8cd-a45ee5a7b3b0/background.html)
Uncaught 
Exception { name: "NS_ERROR_FILE_NOT_FOUND", message: "Component returned failure code: 0x80520012 (NS_ERROR_FILE_NOT_FOUND) [nsIIOService.newChannelFromURI]", result: 2152857618, filename: "resource://gre/modules/NetUtil.jsm", lineNumber: 311, columnNumber: 0, data: null, stack: "NetUtil_newChannel@resource://gre/modules/NetUtil.jsm:311:31\nonLinkAdded@chrome://messenger/content/specialTabs.js:309:32\nhandleEvent@chrome://messenger/content/specialTabs.js:249:14\n", location: XPCWrappedNative_NoHelper }
​
columnNumber: 0
​
data: null
​
filename: "resource://gre/modules/NetUtil.jsm"
​
lineNumber: 311
​
location: XPCWrappedNative_NoHelper { QueryInterface: QueryInterface(), filename: Getter, name: Getter, … }
​
message: "Component returned failure code: 0x80520012 (NS_ERROR_FILE_NOT_FOUND) [nsIIOService.newChannelFromURI]"
​
name: "NS_ERROR_FILE_NOT_FOUND"
​
result: 2152857618
​
stack: "NetUtil_newChannel@resource://gre/modules/NetUtil.jsm:311:31\nonLinkAdded@chrome://messenger/content/specialTabs.js:309:32\nhandleEvent@chrome://messenger/content/specialTabs.js:249:14\n"
​
<prototype>: ExceptionPrototype { toString: toString(), name: Getter, message: Getter, … }
[NetUtil.jsm:311](resource://gre/modules/NetUtil.jsm)
No chrome package registered for chrome://browser/skin/window.svg
Trying to load C:\Program Files\Mozilla Thunderbird\libotr.dll [OTRLib.jsm:64:11](resource:///modules/OTRLib.jsm)
Successfully loaded OTR library C:\Program Files\Mozilla Thunderbird\libotr.dll [OTRLib.jsm:72:13](resource:///modules/OTRLib.jsm)

**Comment by ercek** [comment]
Update: The Tools menu item still does not show up, but I was able to right click on Local Folders and import from there.

**Comment by cleidigh** [comment]
@ercek 
I do all my main development and testing on Win10 and final testing on my MacAir and Linux VM.

Thanks for the console output. I do see several errors that do not look normal,however, I don't think they originate from IETNG.

So are any of the context menus working (right click on message or folder)?
Can you get the options dialog from the wrench button in the extension entry?

If you clear the console then disable the extension, renable and then capture output we should see some startup output.
@cleidigh

**Comment by cleidigh** [comment]
@ercek 
our posts crossed.
So to be clear the tools menu is on the main menu which has to be enabled with Alt-V then toggle toolbar main menu.
@cleidigh

**Comment by ercek** [comment]
Thank you so much! The Alt-V version of the toolbar is the one I had been searching for until I discovered the right click on the folder. 
To answer your questions: 
Yes, right clicking on a message or folder gives me context menus. If I do so on the folder, I see the ImportExport option. I used this to pull in my mbox files. 
I can also see the options dialog using the wrench button when I open the extension. 
The Alt-V/Tools shortcut also lists it.
The only place it isn't appearing is the top right menu button (three bars, next to the search entry textbox) which then has a Tools option as well. This is the only obvious menu when I first open Thunderbird (having not used the program in a decade, I presumed that was what all the instructions meant when they pointed me to the Menu). Hoping this will be useful for others if they end up in the same boat. 
Thanks again for your help on this and for creating this add-on.
